feat: object store to pvc migrations (#633)
replicatedhq/kurl.sh: src/markdown-pages/add-ons/kotsadm.md

The [KOTS add-on](https://kots.io/kotsadm/installing/installing-a-kots-app/) installs an admin console for managing KOTS apps.

> This add-on requires an S3-Compatible object store be available in the cluster by default. 

Reviewer: ```suggestion
By default, this add-on requires an S3-Compatible object store be available in the cluster. 
```

    uiBindPort: 8800
    hostname: "hostname"
    applicationNamespace: "kots"
    disableS3: true
```

flags-table

> ## Side Effects of the `disableS3` flag

Reviewer 2: I wonder if removing the word "Side" here would sound better. Side effects connote something unintentional, so I wonder if "Effects of the `disableS3` flag" would work better.

Reviewer: I like this suggestion!

> Object storage is used by the KOTS addon as default behaviors in the following ways:

Reviewer: ```suggestion
Object storage is used by the KOTS add-on as default behaviors in the following ways:
```

1. Directly, as a way for the console to store support-bundles and previous versions of your application.

> 1. Directly, when hostpath or NFS locations are set in the console. The console will dynamically create a Minio instance and attach to the specified source.

Reviewer: ```suggestion
1. Directly, when hostpath or NFS locations are set in the console. The console will dynamically create a Minio instance and attach it to the specified source.
```

> 1. Indirectly, as part of the Registry addon, to store container images in airgap deployments.

Reviewer: ```suggestion
1. Indirectly, as part of the Registry add-on, to store container images in air gap deployments.
```

> 1. Indirectly, as part of the Velero addon, to store snapshots in the cluster using the "Internal" option.

Reviewer: ```suggestion
1. Indirectly, as part of the Velero add-on, to store snapshots in the cluster using the `Internal` option.
```

> When the disableS3 flag is set to `true`, the KOTS admin console will switch to use a PV for all of the direct operations. 

Reviewer: ```suggestion
When the `disableS3` flag is set to `true`, the KOTS admin console will switch to use a PV for all of the direct operations. 
```

> It will also have the following side effects outside of the KOTS addon:

Reviewer: ```suggestion
It will also have the following side effects outside of the KOTS add-on:
```

> It will also have the following side effects outside of the KOTS add-on:

Reviewer 2: Same comment here, maybe "effects" instead of "side effects".

> 1. For new installs, the Regsistry addon behavior will not change. It will use an object store if it is available, otherwise it will use a PV. For upgrades or re-installs, a migration will be performed from an object store to a PV ONLY if the object store is removed from the installer spec.

Reviewer: ```suggestion
1. For new installs, the Registry add-on behavior will not change. It will use an object store if it is available, otherwise it will use a PV. For upgrades or re-installs, a migration will only be performed from an object store to a PV if the object store is removed from the installer spec.
```

> 1. For new installs, the Regsistry addon behavior will not change. It will use an object store if it is available, otherwise it will use a PV. For upgrades or re-installs, a migration will be performed from an object store to a PV ONLY if the object store is removed from the installer spec.

Author: resolved in comment by Samantha

> 1. For new installs, the Regsistry addon behavior will not change. It will use an object store if it is available, otherwise it will use a PV. For upgrades or re-installs, a migration will be performed from an object store to a PV ONLY if the object store is removed from the installer spec.

Author: it's add-on

> 1. For new installs, the Velero addon will use a PV for storing "Internal" snapshots instead of using object storage. For upgrades or re-installs, internal snapshots will be migrated from the object store to a PV.

Reviewer: ```suggestion
1. For new installs, the Velero add-on will use a PV for storing _Internal_ snapshots instead of using object storage. For upgrades or re-installs, _internal_ snapshots will be migrated from the object store to a PV.
```

> Migrations will NOT be performed if the flag is not set. 

Reviewer: ```suggestion
**Note:** The flag must be set to perform migrations.
```
